src/include/platform/FailSafeContext.h

```diff
@@ -0,0 +1,71 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          A 'Fail Safe Context' SHALL be created on the receiver, to track fail-safe
+ *          state information while the fail-safe is armed.
+ */
+
+#pragma once
+
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+class FailSafeContext
+{
+public:
+    // ===== Members for internal use by other Device Layer components.
+
+    /**
+     * @brief
+     *  Only a single fail-safe timer is started on the device, if this function is called again
+     *  when the fail-safe timer is currently armed, the currently-running fail-safe timer will
+     *  first be cancelled, then the fail-safe timer will be re-armed.
+     */
+    CHIP_ERROR ArmFailSafe(FabricIndex accessingFabric, System::Clock::Timeout expiryLength);
```

Comment:
```suggestion
    CHIP_ERROR ArmFailSafe(FabricIndex accessingFabricIndex, System::Clock::Timeout expiryLength);
```

fabric != fabric index.